File: C++/freeCodeCamp.org_C++ Tutorial for Beginners/src/028_Inheritance/Inheritance.cpp
#include <iostream>

/* Simple C++ program that uses inheritance */

using namespace std;

class Chef // Chef is a superclass
{
    public:
        void makeChicken()
        {
            cout << "The chef makes chicken" << endl;
        }
        void makeSalad()
        {
            cout << "The chef makes salad" << endl;
        }
        void makeSpecialDish()
        {
            cout << "The chef makes bbc ribs" << endl;
        }
};

class ItalianChef : public Chef // inherit the Chef class. ItalianChef is a subclass of the Chef class
{
    public:
        void makePasta()
        {
            cout << "The italian chef makes pasta" << endl;
        }
        void makeSpecialDish() // Overrides the superclasses makeSpecialDish() function
        {
            cout << "The italian chef makes chicken parm" << endl;
        }
};

int main()
{
    Chef chef;
    chef.makeChicken();

    ItalianChef italianChef;
    italianChef.makeChicken();

    chef.makeSpecialDish();
    italianChef.makeSpecialDish();

    return 0;
}

File: Python/002_OOP Basics/028_Inheritance/Inheritance.py
class Chef: # Chef is a superclass

    def makeChicken(self):
        print("The chef makes chicken")
    
    def makeSalad(self):
        print("The chef makes salad")
    
    def makeSpecialDish(self):
        print("The chef makes bbc ribs")

class ItalianChef(Chef): # inherit the Chef class. ItalianChef is a subclass of the Chef class
    
    def makePasta(self):
        print("The italian chef makes pasta")
    
    def makeSpecialDish(self):
        print("The italian chef makes chicken parm")

chef1 = Chef()
chef1.makeChicken()

chef2 = ItalianChef()
chef2.makeChicken()

chef1.makeSpecialDish()
chef2.makeSpecialDish()